R/RunStudy.R

```diff
@@ -518,20 +522,26 @@ loadCohortsFromPackage <- function(cohortIds) {
 loadCohortsForExportFromPackage <- function(cohortIds) {
   packageName = getThisPackageName()
   cohorts <- getCohortsToCreate()
-  cohorts$atlasId <- NULL
+  cohorts <- cohorts %>%  dplyr::mutate(atlasId = NULL)
   if ("atlasName" %in% colnames(cohorts)) {
-    cohorts <- dplyr::rename(cohorts, cohortName = "atlasName", cohortFullName = "atlasName")
+    # Remove PIONEER cohort identifier (3.g. [PIONEER O2])
+    # Remove atlasName and name from object to prevent clashes when combining with stratXref
+    cohorts <- cohorts %>% 
+      dplyr::mutate(cohortName = trimws(gsub("(\\[.+?\\])", "", atlasName)),
+                    cohortFullName = atlasName) %>%
+      dplyr::select(-atlasName, -name)
   } else {
-    cohorts <- dplyr::rename(cohorts, cohortName = "name", cohortFullName = "fullName")
+    cohorts <- cohorts %>% dplyr::rename(cohortName = name, cohortFullName = fullName)
   }
   
   # Get the stratified cohorts for the study
   # and join to the cohorts to create to get the names
-  targetStrataXref <- getTargetStrataXref()
-  targetStrataXref <- dplyr::rename(targetStrataXref, cohortName = "name")
-  targetStrataXref$cohortFullName <- targetStrataXref$cohortName
-  targetStrataXref$targetId <- NULL
-  targetStrataXref$strataId <- NULL
+  targetStrataXref <- getTargetStrataXref() 
+  targetStrataXref <- targetStrataXref %>% 
+    dplyr::rename(cohortName = name) %>%
+    dplyr::mutate(cohortFullName = cohortName,
+                  targetId = NULL,
+                  strataId = NULL)
   
   cols <- names(cohorts)
   cohorts <- rbind(cohorts, targetStrataXref[cols])
```

Comment:
dplyr row binding is more forgiving if columns don't match. Will just fill NA.
```suggestion
  cohorts <- dplyr::bind_rows(cohorts, targetStrataXref)
```